src/codeu/chat/server/Model.java

```diff
@@ -125,4 +131,17 @@ public void add(Message message) {
   public StoreAccessor<String, Message> messageByText() {
     return messageByText;
   }
+  
+  public InterestSet getInterestSet(Uuid id){
+    System.out.println(interestMap.get(id));
+    System.out.println("CURRENT: "+interestMap.size());
+    return interestMap.get(id);
+  }
+  public void updateInterests(Uuid id, InterestSet intSet){
+    System.out.println("BEFORE: "+interestMap.size());
+    interestMap.remove(id);
```

Comment:
You don't need to call remove here. `interestMap.put` alone will replace whatever was in that object's position in the map.